src/Microsoft.AspNet.OData.Shared/Builder/ParameterConfiguration.cs

```diff
@@ -51,5 +51,44 @@ protected ParameterConfiguration(string name, IEdmTypeConfiguration parameterTyp
         /// Gets or sets a value indicating whether this parameter is nullable or not.
         /// </summary>
         public bool Nullable { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether this parameter is optional or not.
+        /// </summary>
+        public bool IsOptional { get; set; }
+
+        /// <summary>
+        /// Gets or sets a default value for optional parameter.
+        /// </summary>
+        public string DefaultValue { get; set; }
```

Comment:
Should we do a private set? HasDefaultValue performs the set for us and couples it with setting IsOptional = true.

Reply:
Yes. Thanks. I changed it to use "protected set".